vdr/types/interface.go

```diff
@@ -29,15 +29,20 @@ import (
 	"github.com/nuts-foundation/nuts-node/crypto/hash"
 )
 
-// DocResolver is the interface that groups all the DID Document read methods
-type DocResolver interface {
+// DIDResolver is the interface for DID resolvers: the process of getting the backing document of a DID.
+type DIDResolver interface {
 	// Resolve returns a DID Document for the provided DID.
 	// If metadata is not provided the latest version is returned.
 	// If metadata is provided then the result is filtered or scoped on that metadata.
 	// It returns ErrNotFound if there are no corresponding DID documents or when the DID Documents are disjoint with the provided ResolveMetadata
 	// It returns ErrDeactivated if the DID Document has been deactivated and metadata is unset or metadata.AllowDeactivated is false.
 	// It returns ErrNoActiveController if all of the DID Documents controllers have been deactivated and metadata is unset or metadata.AllowDeactivated is false.
 	Resolve(id did.DID, metadata *ResolveMetadata) (*did.Document, *DocumentMetadata, error)
+}
+
+// DocResolver is the interface that groups all the DID Document read methods.
+type DocResolver interface {
```

Comment:
`ResolveControllers` could just be a function. It's only used for the Nuts method or nested from a resolver. It's used for mocking as well, but that could just be replaced with a resolve mock with the same outcome?

Reply:
Comment due to having DIDResolver and DocResolver for which you can't deduce from the name what they do or what the difference is.

Reply:
DocResolver is renamed to `didnuts.DIDResolver` in a following PR